python_modules/libraries/dagster-components/dagster_components/core/component_decl_builder.py

```diff
@@ -124,7 +101,7 @@ def from_path(path: Path) -> "YamlComponentDecl":
 
     def get_component_type(self, registry: ComponentTypeRegistry) -> type[Component]:
         parsed_defs = self.component_file_model
-        key = ComponentKey.from_typename(parsed_defs.type)
+        key = ComponentKey.from_typename(parsed_defs.type, self.path)
         if parsed_defs.type.endswith(".py"):
```

Comment:
As a follow up we should canonicalize `endswith(".py")` and have that be in a single spot in the codebase.

Reply:
pushed this logic into the registry